frontend/src/components/maps/providers/PropertyPopUpProvider.tsx

```diff
@@ -0,0 +1,25 @@
+import { noop } from 'lodash';
```

Comment:
This one is hard to explain, so let me just provide some code:

```
import { noop } from 'lodash';
import React from 'react';
import { IParcel, IBuilding } from 'actions/parcelsActions';

interface IPopupContext {
  propertyInfo: IParcel | IBuilding | null;
  setPropertyInfo: (propertyInfo: IParcel | IBuilding | null) => void;
  propertyTypeID: number | null;
  setPropertyTypeID: (propertyTypeId: number) => void;
}

export const PropertyPopUpContext = React.createContext<IPopupContext>({
  propertyInfo: null,
  setPropertyInfo: noop,
  propertyTypeID: null,
  setPropertyTypeID: noop,
});

/**
 * Allows for the property information to be sent to the map
 * when the user clicks on a marker
 */
export const PropertyPopUpContextProvider: React.FC = ({ children }) => {
  const [propertyInfo, setPropertyInfo] = React.useState<IParcel | IBuilding | null>(null);
  const [propertyTypeID, setPropertyTypeID] = React.useState<number | null>(null);
  return (
    <PropertyPopUpContext.Provider
      value={{ propertyInfo, setPropertyInfo, propertyTypeID, setPropertyTypeID }}
    >
      {children}
    </PropertyPopUpContext.Provider>
  );
};

```

Reply:
by using generics, you can provide the expected typings for context and state hooks, which make things much cleaner.

Reply:
If you don't do this, typescript/react infers the type based on the initialized value, which is null in this case. We don't want our context fields typings to be null.

Reply:
Thank you so much, that's really helpful